views/editHome.js

```diff
@@ -0,0 +1,207 @@
+
+class Row extends Component {
+  static propTypes = {
+    active: React.PropTypes.bool,
+    data: React.PropTypes.object.isRequired,
+  }
+
+  state = {
+    style: {
+      shadowRadius: new Animated.Value(2),
+      transform: [{scale: new Animated.Value(1)}],
+    },
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.active !== nextProps.active) {
+      if (nextProps.active) {
+        this.startActivationAnimation()
+      } else {
+        this.startDeactivationAnimation()
+      }
+    }
+  }
+
+  props: {
+    data: ViewType,
+    active: boolean,
+  };
+
+  startActivationAnimation = () => {
+    const {style} = this.state
+    Animated.parallel([
+      Animated.timing(style.transform[0].scale, {
+        duration: 100,
+        easing: Easing.out(Easing.quad),
+        toValue: 1.05,
+      }),
+      Animated.timing(style.shadowRadius, {
+        duration: 100,
+        easing: Easing.out(Easing.quad),
+        toValue: 10,
+      }),
+    ]).start()
+  }
+
+  startDeactivationAnimation = () => {
+    const {style} = this.state
+    Animated.parallel([
+      Animated.timing(style.transform[0].scale, {
+        duration: 100,
+        easing: Easing.out(Easing.quad),
+        toValue: 1,
+      }),
+      Animated.timing(style.shadowRadius, {
+        duration: 100,
+        easing: Easing.out(Easing.quad),
+        toValue: 2,
+      }),
+    ]).start()
+  }
+
+  render() {
+    return (
+      <Animated.View style={[styles.row, this.state.style]}>
+        <MenuIcon icon={this.props.data.icon} tint={this.props.data.tint} />
+        <Text style={[styles.text, {color: this.props.data.tint}]}>
+          {this.props.data.title}
+        </Text>
+        <ReorderIcon />
+      </Animated.View>
+    )
+  }
+}
+
+export default class EditHomeView extends React.Component {
+  static propTypes = {
+    navigator: React.PropTypes.instanceOf(Navigator),
+    route: React.PropTypes.object,
+  }
+
+  state = {
+    loaded: false,
+    order: Object.keys(objViews),
+  }
+
+  componentWillMount() {
+    this.loadData()
+  }
+
+  renderRow({data, active}: {data: ViewType, active: boolean}) {
+    return <Row data={data} active={active} />
+  }
+
+  loadData = async () => {
+    this.setState({loaded: false})
+
+    let savedOrder = JSON.parse(await AsyncStorage.getItem('homescreen:view-order'))
+
+    // check to see if we have a modified view order or not
+    savedOrder = savedOrder || Object.keys(objViews)
+
+    this.setState({loaded: true, order: savedOrder})
+  }
+
+  onOrderChange = (order: {[key: string]: ViewType}) => {
+    AsyncStorage.setItem('homescreen:view-order', JSON.stringify(Object.values(order)))
+  }
+
+  render() {
+    if (!this.state.loaded) {
+      return <LoadingView />
+    }
+
+    return (
+      <SortableList
+        contentContainerStyle={styles.contentContainer}
+        data={objViews}
+        order={this.state.order}
+        onChangeOrder={this.onOrderChange}
```

Comment:
I wonder if it would be better to store the order of the items in memory, then call the AsyncStorage stuff when the user hits the done button? Seems unnecessary to be writing to storage every time they change one when they only see the result when they hit done. (I might be missing a very good reason for it to work this way!)